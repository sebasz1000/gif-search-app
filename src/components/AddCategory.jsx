import { useState } from "react";
import PropTypes from "prop-types";
export const AddCategory = ({ onAddCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const value = inputValue.trim();
    if (value.length < 2) return;

    onAddCategory(value);
    setInputValue("");
  };

  return (
    <form aria-label="form" onSubmit={handleSubmit}>
      <input
        type="text"
        value={inputValue}
        placeholder="Search Gifs"
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onAddCategory: PropTypes.func.isRequired,
};
