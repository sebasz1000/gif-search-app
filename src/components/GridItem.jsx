import PropTypes from "prop-types";
export const GridItem = ({ id, title, url }) => {
  return (
    <div key={id} className="card">
      <h4 className="text-white font-weight-light">{title}</h4>
      <img src={url} alt={title} />
    </div>
  );
};

GridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
