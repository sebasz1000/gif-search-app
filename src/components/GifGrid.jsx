import { GridItem } from "./GridItem";
import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
//import loaderUrl from "./../assets/loader.gif";
export const GifGrid = ({ category }) => {
  const { gifs, isLoading } = useFetchGifs(category);

  return (
    <>
      <section>
        <h2 className="text-center text-capitalize text-muted">{category}</h2>

        {isLoading && (
          /*<img
            aria-label="loader"
            className="loader"
            src={loaderUrl}
            width="50px"
            alt="Loader image"
          /> PUT ME BACK AFTER TESTING*/
          <h4 className="text-white text-center">Loading...</h4>
        )}
        {gifs.length < 0 && !isLoading ? (
          <p className="text-muted">
            No results were found for <strong>"{category}"</strong>
          </p>
        ) : (
          <GridCategory gifs={gifs} />
        )}
      </section>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
const GridCategory = ({ gifs }) => {
  return (
    <section className="card-grid">
      {gifs.map((gif) => {
        return <GridItem {...gif} key={gif.id} />;
      })}
    </section>
  );
};
