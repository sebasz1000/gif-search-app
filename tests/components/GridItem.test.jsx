import { render, screen } from "@testing-library/react";
import { GridItem } from "../../src/components/GridItem";
describe("Tests <GridItem />", () => {
  const name = "Ash",
    url =
      "https://media1.giphy.com/media/G9qfCvxlwGAaQ/giphy.gif?cid=cc5384774mwbhb1qqlv1fa6vi6l0dr6dszwjdwci3ceta1va&rid=giphy.gif&ct=g";
  test("Should match with snapshot", () => {
    const { container } = render(<GridItem id={name} title={name} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Should show image with specified url and alt text", () => {
    render(<GridItem id={name} title={name} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(name);
    //screen.debug();
  });

  test("Should display components title", () => {
    render(<GridItem id={name} title={name} url={url} />);
    expect(screen.getByText(name)).toBeTruthy();
  });
});
