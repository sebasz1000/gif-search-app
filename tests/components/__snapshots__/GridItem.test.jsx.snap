// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests <GridItem /> Should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <h4
      class="text-white font-weight-light"
    >
      Ash
    </h4>
    <img
      alt="Ash"
      src="https://media1.giphy.com/media/G9qfCvxlwGAaQ/giphy.gif?cid=cc5384774mwbhb1qqlv1fa6vi6l0dr6dszwjdwci3ceta1va&rid=giphy.gif&ct=g"
    />
  </div>
</div>
`;
